src/Microsoft.Health.Fhir.Azure/ExportDestinationClient/OrderedSetOfBlockIds.cs

```diff
@@ -0,0 +1,62 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using EnsureThat;
+
+namespace Microsoft.Health.Fhir.Azure.ExportDestinationClient
+{
+    /// <summary>
+    /// A modified implementation of an ordered hash set using a linked list and hashset. This class is specifically used
+    /// for maintaining the existing block ids that are part of a blob in <see cref="CloudBlockBlobWrapper"/>. We need to
+    /// maintain ordering of the block ids but also make sure that we don't have multiple copies of a block id. Hence
+    /// the need for an ordered hash set.
+    /// </summary>
+    internal class OrderedSetOfBlockIds
```

Comment:
I think you could use built-in SortedSet<T> since we are using the page number as the id.

Reply:
We won't be able to use SortedSet here because we are base64 encoding the page number into a string. And in that case SortedSet will alter the order. If we change the way we generate the blockId, we can modify this part accordingly.